ocaml/xapi/xapi.ml

```diff
@@ -542,6 +542,26 @@ let check_network_reset () =
          let gateway = if List.mem_assoc "GATEWAY" args then List.assoc "GATEWAY" args else "" in
          let dNS = if List.mem_assoc "DNS" args then List.assoc "DNS" args else "" in
 
+         (* Get the existing network for management vlan *)
+         let existing_network =
+           match vlan with
+           | Some vlan ->
+             (* Check if emergency network reset is on slave then use the pool management vlan network while creating vlan *)
+             if Pool_role.is_slave () then
+               let management_pif = Xapi_host.get_management_interface ~__context ~host:(Helpers.get_master ~__context) in
+               Some (Db.PIF.get_network ~__context ~self:management_pif)
+             else begin
+               match Db.VLAN.get_refs_where ~__context ~expr:(Eq (Field "tag", Literal vlan)) with
```

Comment:
This may get you a VLAN on a different physical PIF. It's better to search for PIFs with matching `VLAN` and `device`. And then that should also work for slaves, so I don't think we need the conditional here?